python/cugraph_alg.py

```diff
@@ -0,0 +1,56 @@
+import mgp
+import pandas
+import sys
+
+try:
+    import cugraph
+except ImportError as import_error:
+    sys.stderr.write(
+        "RAPIDS cuGraph is not installed, please build Memgraph from source with Python Conda interpreter where cuGraph is installd ."
+    )
+    raise import_error
+
+
+class GraphDefinition:
+    SOURCE = "source"
+    TARGET = "target"
+
+    VERTEX = "vertex"
+    PAGERANK = "pagerank"
+
+
+def _get_mg_data(mg_graph: mgp.Graph) -> pandas.DataFrame:
+    graph_data = {GraphDefinition.SOURCE: [], GraphDefinition.TARGET: []}
+    for from_vertex in mg_graph.vertices:
+        for edge in from_vertex.out_edges:
+            to_vertex = edge.to_vertex
+            graph_data[GraphDefinition.SOURCE].append(from_vertex.id)
+            graph_data[GraphDefinition.TARGET].append(to_vertex.id)
+
+    df = pandas.DataFrame(graph_data)
```

Comment:
Rather than appending to `list`, try generating in a stream-like manner to `pandas.DataFrame`